src/morale.cpp

```diff
@@ -781,22 +789,33 @@ void player_morale::update_constrained_penalty()
 void player_morale::update_squeamish_penalty()
 {
     int penalty = 0;
-    const auto bp_pen = [ this ]( body_part bp, int penalty ) -> int {
-        return (
-            body_parts[bp].filthy > 0 ||
-            body_parts[opposite_body_part( bp )].filthy > 0 ) ? penalty : 0;
-    };
-    penalty = ( bp_pen( bp_torso,  6 ) +
-                bp_pen( bp_head,   7 ) +
-                bp_pen( bp_eyes,   8 ) +
-                bp_pen( bp_mouth,  9 ) +
-                bp_pen( bp_leg_l,  5 ) +
-                bp_pen( bp_leg_r,  5 ) +
-                bp_pen( bp_arm_l,  5 ) +
-                bp_pen( bp_arm_r,  5 ) +
-                bp_pen( bp_foot_l, 3 ) +
-                bp_pen( bp_foot_r, 3 ) +
-                bp_pen( bp_hand_l, 3 ) +
-                bp_pen( bp_hand_r, 3 ) );
+
+    if( !nonsqueamish ) {
```

Comment:
In terms of pseudocode I view it like this:
`!nonsqueamish == squeamish /**(strong squeamish)*/ || ordinary /**(soft squeamish)*/`